ibis/backends/trino/__init__.py

```diff
@@ -354,53 +454,123 @@ def literal_compile(v):
-                    f"ALTER TABLE IF EXISTS {table_ref} RENAME TO {self._quote(orig_table_ref)}"
+                cur.execute(
+                    sge.AlterTable(
+                        this=table_ref,
+                        exists=True,
+                        actions=[sge.RenameTable(this=orig_table_ref, exists=True)],
+                    ).sql(self.name)
                 )
 
-        return self.table(orig_table_ref)
+        return self.table(orig_table_ref.name)
 
-    def _table_from_schema(
-        self,
-        name: str,
-        schema: sch.Schema,
-        temp: bool = False,
-        database: str | None = None,
-        **kwargs: Any,
-    ) -> sa.Table:
-        return super()._table_from_schema(
-            name,
-            schema,
-            temp=temp,
-            trino_catalog=database or self.current_database,
-            **kwargs,
+    def _get_temp_view_definition(self, name: str, definition: str) -> str:
+        return sge.Create(
+            this=sg.to_identifier(name, quoted=self.compiler.quoted),
+            kind="VIEW",
+            expression=definition,
+            replace=True,
         )
+
+    def _register_temp_view_cleanup(self, name: str) -> None:
+        def drop(self, name: str, query: str):
+            self.raw_sql(query)
+            self._temp_views.discard(name)
+
+        query = sge.Drop(this=sg.table(name), kind="VIEW", exists=True)
+        atexit.register(drop, self, name=name, query=query)
+
+    def _fetch_from_cursor(self, cursor, schema: sch.Schema) -> pd.DataFrame:
+        import pandas as pd
+
+        from ibis.backends.trino.converter import TrinoPandasData
+
+        try:
+            df = pd.DataFrame.from_records(
+                cursor.fetchall(), columns=schema.names, coerce_float=True
+            )
+        except Exception:
+            # clean up the cursor if we fail to create the DataFrame
+            #
+            # in the sqlite case failing to close the cursor results in
+            # artificially locked tables
+            cursor.close()
+            raise
+        df = TrinoPandasData.convert_table(df, schema)
+        return df
+
+    def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
+        schema = op.schema
+        if null_columns := [col for col, dtype in schema.items() if dtype.is_null()]:
+            raise com.IbisTypeError(
+                "Trino cannot yet reliably handle `null` typed columns; "
+                f"got null typed columns: {null_columns}"
+            )
+
+        # only register if we haven't already done so
+        if (name := op.name) not in self.list_tables():
+            quoted = self.compiler.quoted
+            column_defs = [
+                sg.exp.ColumnDef(
+                    this=sg.to_identifier(colname, quoted=quoted),
+                    kind=self.compiler.type_mapper.from_ibis(typ),
+                    # constraints=(
+                    #     None
```

Comment:
Can this be deleted?

Reply:
Yep, done. Also added a comment about why we're not setting `constraints`.